### ISSUE TITLE
[ENH] Support unequal length for rocket transformers

### ISSUE THREAD

**Issue by baraline** [aeon-toolkit/aeon#2449]
### Describe the feature or idea you want to propose

We previously had this feature, but it has gone away during the refactoring of the many versions of the rocket transformers.

### Describe your proposed solution

Implement the functionality back with using numba Typed List as X type. Would need to benchmark if it is worth doing another version for unequal length, or if the adaptation doesn't harm performance significatively.

**Comment by baraline** [comment]
@maorgranot1 for tracking

**Comment by TonyBagnall** [comment]
See 
https://github.com/aeon-toolkit/aeon/pull/2351

The problem comes when the shortest series in transform is shorter than the shortest seem in fit. In this case the convolution overruns the the transform data causing  and out of bounds exception. so if you remember we had a long discussion about this and did not agree on how best to do it. This issue effects RDST and to a lesser extent the shapelet transform too. 

Solutions considered and rejected in discussion were
1) throw an exception: this would happen approximately 50% of the time, so is not really exceptional
2) Refit kernels in transform: this breaks the basic "transform does not change estimator model" and is confusing
3) Shorten over long dilations/kernels in transform. Also considered unacceptable as basically unexpected behaviour

basically its an algorithmic component on how to do this and the there is no published way of doing this. The legacy multivariate/minirocket approach involved concatenation over channels and zero padding, was more arbitrary than the above solutions.

**Comment by baraline** [comment]
Seems like a resonable strategy, the other alternative are not very satisfying anyway. Throwing a warning when this happens to notify the user of it happening in the background and this will do it IMO

**Comment by baraline** [comment]
@TonyBagnall just to confirm, we create kernels based on the shortest series in fit then if we pad to shortests during transform?

**Comment by TonyBagnall** [comment]
@baraline yes, that is what angus suggested

**Comment by angus924** [comment]
Sorry for the slow reply. Potentially you could use shortest, median, mean, etc, length when fitting, and then pad to that length on transform.  In many cases it may not make much difference, but bigger picture it probably makes more sense to use something like median length, in case the minimum length in the training set is unrepresentative.  But I don't have a strong opinion here.

**Comment by baraline** [comment]
It could be a parameter which default on median length then (for example). I think it would make sense to have a new estimator, separate from equal length version, if we go that way.

**Comment by MatthewMiddlehurst** [comment]
No strong feelings on this, but I still feel just returning NaN for kernels which don't fit is the correct choice. We have always treated padding for stuff like this as a pre-processing step unless it is explicitly part of the algorithms, and I don't really see why that would be different here.

If consensus differs from this I have no issue with that going forward, however.
